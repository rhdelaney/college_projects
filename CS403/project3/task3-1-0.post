
(define (lusth-comp a b) 
    (> (string-compare a b) 0)
    )
(define (lusth-eq a b)
    (equal? a b)
    )
(println "creating the tree")
(define t (bst lusth-comp))
(println "doing a bunch of inserts")
(t 'insert "ibsr" 10)
(t 'insert "gyhi" 9)
(t 'insert "vxav" 8)
(t 'insert "wcrz" 7)
(t 'insert "umdu" 6)
(t 'insert "kxeg" 5)
(t 'insert "ecvq" 4)
(t 'insert "ldpt" 3)
(t 'insert "pteh" 2)
(t 'insert "hwrl" 1)
(inspect (t 'size))
(inspect (t 'walk))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'find "pteh" lusth-eq))
(inspect (t 'find "ecvq" lusth-eq))
(inspect (t 'find "zfpb" lusth-eq))
(inspect (t 'find "drpa" lusth-eq))
(inspect (t 'find "gyhi" lusth-eq))
(inspect (t 'find "vdgd" lusth-eq))
(inspect (t 'find "gclr" lusth-eq))
(t 'delete "hwrl" lusth-eq)
(t 'delete "ecvq" lusth-eq)
(t 'delete "kxeg" lusth-eq)
(inspect (t 'walk))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'next))
(inspect (t 'find "hwrl" lusth-eq))
(inspect (t 'find "ldpt" lusth-eq))
(inspect (t 'find "umdu" lusth-eq))
(inspect (t 'find "xhwu" lusth-eq))
(inspect (t 'find "wadf" lusth-eq))
(inspect (t 'find "iegc" lusth-eq))
(inspect (t 'find "roir" lusth-eq))
