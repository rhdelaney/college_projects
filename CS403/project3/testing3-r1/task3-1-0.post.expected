creating the tree
doing a bunch of inserts
(t (quote size)) is 10
(t (quote walk)) is 7
(t (quote next)) is 8
(t (quote next)) is 6
(t (quote next)) is 2
(t (quote next)) is 3
(t (quote next)) is 5
(t (quote next)) is 10
(t (quote next)) is 1
(t (quote next)) is 9
(t (quote next)) is 4
(t (quote next)) is nil
(t (quote find) "pteh" lusth-eq) is 2
(t (quote find) "ecvq" lusth-eq) is 4
(t (quote find) "zfpb" lusth-eq) is nil
(t (quote find) "drpa" lusth-eq) is nil
(t (quote find) "gyhi" lusth-eq) is 9
(t (quote find) "vdgd" lusth-eq) is nil
(t (quote find) "gclr" lusth-eq) is nil
(t (quote walk)) is 7
(t (quote next)) is 8
(t (quote next)) is 6
(t (quote next)) is 2
(t (quote next)) is 3
(t (quote next)) is 10
(t (quote next)) is 9
(t (quote next)) is nil
(t (quote next)) is nil
(t (quote next)) is nil
(t (quote next)) is nil
(t (quote find) "hwrl" lusth-eq) is nil
(t (quote find) "ldpt" lusth-eq) is 3
(t (quote find) "umdu" lusth-eq) is 6
(t (quote find) "xhwu" lusth-eq) is nil
(t (quote find) "wadf" lusth-eq) is nil
(t (quote find) "iegc" lusth-eq) is nil
(t (quote find) "roir" lusth-eq) is nil
